go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -544,16 +544,38 @@ func (c *Controller) RequestFullSync(nodeInfos []nodeInfo) {
 
 // handlers
 
+// skipService is used when UDN is enabled to know which services are to be skipped because they don't
+// belong to the network that this service controller is responsible for.
+func (c *Controller) skipService(name, namespace string) bool {
+	if util.IsNetworkSegmentationSupportEnabled() {
+		serviceNetwork, err := util.GetActiveNetworkForNamespace(namespace, c.nadLister)
```

Comment:
so this GetActiveNetworkForNamespace iterates through all NADs in a for loop to try to find the one. This is going to be very slow with how many services we process. I guess its not too bad because its namespace scoped lookup, and there shouldn't be that many networks in a namespace. In the future lets make sure we have a github issue/jira or something to track making a controller to track networks mapping to namespaces.